### PR TITLE
porting documentation changes from series/1.x to series/2.x

### PR DESCRIPTION
**Pull request by googley42**

### CODE REVIEW

**Comment by googley42** [comment]
@khajavi can you please review this PR as it ports over your documentation changes for the ZIO1 `master` branch to the new `series/2.x` branch - the intention is to make this the default branch after this PR is merged.

I got around the CI build problems by preserving your memory options in `ci.yml`

File: .gitignore [hunk 1]
```diff
@@ -464,6 +464,7 @@ project/metals.sbt
 # mdoc
 website/node_modules
 website/build
+website/yarn.lock
```
Comment (khajavi) [inline .gitignore]: Why ignore yarn.lock?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -46,10 +46,33 @@ jobs:
...
+      - name: Checkout current branch
+        uses: actions/checkout@v2.4.0
+      - name: Setup Scala and Java
+        uses: olafurpg/setup-scala@v13
+      - name: Setup GPG
+        uses: olafurpg/setup-gpg@v3
+      - name: Cache scala dependencies
+        uses: coursier/cache-action@v6
+      - name: Check Document Generation
+        run: ./sbt compileDocs
```
Comment (khajavi) [inline .github/workflows/ci.yml]: `sbt docs/compileDocs`

File: project/plugins.sbt [hunk 3]
```diff
@@ -6,4 +6,6 @@ addSbtPlugin("org.scalameta"      % "sbt-mdoc"       % "2.2.16")
 addSbtPlugin("ch.epfl.scala"      % "sbt-bloop"      % "1.4.6")
 addSbtPlugin("com.eed3si9n"       % "sbt-unidoc"     % "0.4.3")
 addSbtPlugin("com.geirsson"       % "sbt-ci-release" % "1.5.5")
-addSbtPlugin("ch.epfl.scala"      % "sbt-scalafix"   % "0.9.34")
+addSbtPlugin("dev.zio"            % "zio-sbt-website" % "0.1.5+28-4b5b0374-SNAPSHOT")
```
Comment (khajavi) [inline project/plugins.sbt:9]: May I ask you to use the latest version of `zio-sbt-website`, currently `0.3.4`?

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
+---
+id: index
+title: "Overview"
```
Comment (khajavi) [inline docs/index.md]: To make a consistent title with other projects:

title: Introduction to ZIO DynamoDB
sidebar_title: ZIO DynamoDB

File: docs/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+## Add the dependency to your build.sbt file
+
+```scala
+libraryDependencies += "dev.zio" %% "zio-dynamodb" % @VERSION@
+```
+
+### Read & write data to/from DynamoDB
+
+```scala
```
Comment (khajavi) [inline docs/getting-started.md:14]: Not necessary, but it's good to have compile-time safety on code blocks in the documentation, e.g. https://scalameta.org/mdoc/docs/modifiers.html#compile-only
Don't rush, we can do that in a separate PR.

**Comment by khajavi** [comment]
> @khajavi can you please review this PR as it ports over your documentation changes for the ZIO1 `master` branch to the new `series/2.x` branch - the intention is to make this the default branch after this PR is merged.
> 
> I got around the CI build problems by preserving your memory options in `ci.yml`

@googley42 Thank you for the nice work. I added some comments.

File: .gitignore (hunk 1 above)
Comment (googley42) [reply .gitignore]: this was just copied from the master branch  - I will remove

**Comment by khajavi** [comment]
@googley42  Don't move to the latest version. I will make another PR for that.

File: docs/getting-started.md (hunk 5 above)
Comment (googley42) [reply docs/getting-started.md:14]: cool, lets do this in a separate PR then

File: project/plugins.sbt (hunk 3 above)
Comment (googley42) [reply project/plugins.sbt:9]: so I tried this and got a build failure - maybe the binary has not been published?

**Comment by googley42** [comment]
@khajavi thanks for the review. I have addressed the comments - could you re-review please

File: project/plugins.sbt (hunk 3 above)
Comment (khajavi) [reply project/plugins.sbt:9]: I saw that you used it correctly but it needs some migration that is not documented.

**Review by khajavi — APPROVED** [review]

**Comment by googley42** [comment]
@khajavi  thanks for the review - after merging I am going to make this the default branch
